clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -647,6 +648,43 @@ void SemaHLSL::emitLogicalOperatorFixIt(Expr *LHS, Expr *RHS,
       << NewFnName << FixItHint::CreateReplacement(FullRange, OS.str());
 }
 
+void SemaHLSL::handleRootSignatureAttr(Decl *D, const ParsedAttr &AL) {
+  using namespace llvm::hlsl::root_signature;
+  using namespace clang::hlsl;
```

Comment:
These are usually places at the top of the file after `#include`s. If this is the only function where it is needed it might be cleaner to just use fully qualified names.